### ISSUE TITLE
[core] : Widgets not displaying correctly when sent from upgraded 4.1.1 to clean 4.1.1

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#12242]
Reported via Support: https://my.dotcms.com/tickets/detail.dot?id=c92d6f2c-94fb-439c-89af-d13da3321969
Reproducible in dotCMS 4.1.1

When setting up the following Push Publish configuration:

1. **Sender instance:** A clean dotCMS 3.7.1 upgraded to version 4.1.1.
1. **Receiver instance:** A clean dotCMS 4.1.1.

Then, creating a Simple Widget content in the Sender and pushing to the Receiver, the widget is not displaying any information at all when added to any HTML Page. Such a contentlet can be added to a page in both Sender and Receiver instances, and it can be seen in the **Content Search** page. However, AFTER pushing the widget contentlet (which also sends the Simple Widget content type as dependency) it is displayed correctly in the Sender instance ONLY, not the Receiver.

## Expected Behavior
Pushing a widget content created in an upgraded dotCMS 4.1.1 to a clean 4.1.1 instance should be correct and not cause any problems.

## Current Behavior
The content in the Receiver instance is corrupted and is not displaying any information when added to an HTML Page.

## Possible Solution
Review the values of the fields of the Simple Widget content type.

## Known Workarounds
N/A

## Steps to Reproduce
1. Start up a dotCMS 3.7.1 and upgrade it to 4.1.1. This will be the sender.
2. Start another instance with a clean dotCMS 4.1.1. This will be the receiver.
3. Create a Simple Widget contentlet in the sender. Make sure it can be added and displayed on a page.
4. Set up the Push Publish configuration for both servers.
5. Push the contentlet to the receiver.
6. In the receiver, go to the **Content Search** page and make sure the contentlet is there.
7. In the receiver, go to an existing page (or create a new one) and add the widget contentlet you just received.
8. Notice how the widget is not displaying anything on the page.

## Context
All widget contentlets must be pusheable between dotCMS instances with the same version, even when one of the servers was upgraded from a pervious version.

## Your Environment
* dotCMS version used: dotCMS 3.7.1 and 4.1.1
* Browser Name and version: Not relevant.
* Operating System and version: Not relevant.
* Application Server and version: Not relevant.
* Java Brand and version: Java 8
* Database and version: Not relevant.
* Application module (if apply): Push Publishing, Content Type.

**Comment by jwyang99** [comment]
I found a work around, push simple widget content type from the fresh new v4.1.1 instance to the upgraded instance.  Simple Widget field will be replaced with the new constant type instead of hidden.

**Comment by condescendent** [comment]
Good findings! I will upload the picture showing the property for the simple widget in the source is different than it in the destination. The left side is the original server we upgraded from 3.7.1
![simplewidgetpropertydifference](https://user-images.githubusercontent.com/16570859/28898367-69b86e7e-779a-11e7-826a-949e7b5dff3b.png)
, the right side is a new server we installed from fresh with 4.1.1.
